### DISCUSSION TITLE
이벤트 핸들러에서 this는 무엇을 가리키나요?

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#99] [this]
- 이벤트 핸들러에서 this는 무엇을 가리키나요?

**Comment by publdaze** [comment]
> * 이벤트 핸들러에서 this는 무엇을 가리키나요?

이벤트 핸들러 어트리뷰터 방식
- 인수로 this를 넘겨줄 경우는 이벤트를 바인딩한 DOM 요소
- this를 안 넘겨줄 경우는 일반함수로서 전역 객체

이벤트 핸들러 프로퍼티 방식, addEventListener 메서드 방식
- 바인딩한 DOM 요소
- 화살표 함수 사용 시 상위 스코프의 this

**Comment by pipisebastian** [comment]
> * 이벤트 핸들러에서 this는 무엇을 가리키나요?

- 이벤트 핸들러 어트리뷰트
    - HTML 요소의 인라인 형식인 어트리뷰트 방식은, 일반 함수로 호출되어, 전역객체를 바인딩한다.
        
        ```jsx
        <button onClick=”handleClick()”>버튼<button/>
        ```
        
    - 만약 인수로 this를 전달할 경우, 이벤트를 바인딩할 DOM요소를 가리킨다
        
        ```jsx
        <button onClick=”handleClick(this)”>버튼<button/>
        ```
        
- 이벤트 핸들러 프로퍼티, addEventListener 메서드
    - 이벤트를 바인딩할 DOM 요소를 가리킨다
        - 이벤트 핸들러 프로퍼티
            
            ```jsx
            $button1.onClick = function(e){ // 
            	console.log(this === e.currentTarget); // true
            }
            ```
            
        - addEventListener 메서드
            
            ```jsx
            $button1.addEventListener('click', function (e) {
            	console.log(this === e.currentTarget); // true
            })
            ```
            
        - 그러나, 만약 화살표 함수를 사용할 경우엔 this가 전역객체이다.
